packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -388,13 +415,17 @@ export async function downloadDocsAndUpsertFHIR({
 
   // Get File info from S3 (or from memory, if override = true)
   const getFilesWithStorageInfo = async () =>
-    override
+    forceDownload
       ? await Promise.all(validDocs.map(convertToNonExistingS3Info(patient)))
       : await getS3Info(validDocs, patient);
 
   // Get all DocumentReferences for this patient + File info from S3
   const [foundOnFHIR, filesWithStorageInfo] = await Promise.all([
-    getAllPages(() => fhirApi.searchResourcePages("DocumentReference", `patient=${patient.id}`)),
+    ignoreDocRefOnFHIRServer
+      ? []
+      : getAllPages(() =>
+          fhirApi.searchResourcePages("DocumentReference", `patient=${patient.id}`)
+        ),
```

Comment:
Setting `ignoreDocRefOnFHIRServer` to `true` means we'll process all doc refs for the given patient. For those doc refs that have the same ID for all doc ref queries, it means we should be able to reuse the document we downloaded and converted previously, but for those that have an ID generated per request we'll very likely re-download/re-convert them.